Sources/Networking/Core/Upload/UploadAPIManaging.swift

```diff
@@ -73,29 +45,46 @@ public protocol UploadAPIManaging {
 }
 
 public extension UploadAPIManaging {
-    /// Initiates a `multipart/form-data` upload request to the specified `endpoint`.
-    ///
-    /// If the size of the `MultipartFormData` exceeds 10MB, the data is uploaded from disk rather than being loaded into memory all at once. This can help reduce memory usage when uploading large amounts of data.
-    /// To specify different data threshold, use ``upload(multipartFormData:sizeThreshold:to:)``.
-    ///
+    /// Initiates a data upload request for the specified endpoint.
     /// - Parameters:
-    ///   - multipartFormData: The multipart form data to upload.
-    ///   - endpoint: The API endpoint to where data will be sent.
-    ///
+    ///   - type: The data to send to the server.
+    ///   - uploadURL: The URL where data will be sent.
     /// - Returns: An `UploadTask` that represents this request.
-    func upload(
-        multipartFormData: MultipartFormData,
-        to endpoint: Requestable
-    ) async throws -> UploadTask {
-        try await upload(
-            multipartFormData: multipartFormData,
-            sizeThreshold: 10_000_000,
-            to: endpoint
-        )
+    func upload(_ type: UploadType, to uploadURL: URL) async throws -> UploadTask {
+        try await upload(type, to: UploadRouter(url: uploadURL, uploadType: type))
     }
-    
+
     /// Returns an active ``UploadTask`` specified by its identifier.
     func task(with id: UploadTask.ID) async -> UploadTask? {
         await activeTasks.first { $0.id == id }
     }
 }
+
+/// A Router for basic use case of uploading file/data/multiPartForm to a given URL.
+private struct UploadRouter: Requestable {
```

Comment:
I have similar feeling here as in the downloading managing (private struct in public interface)